### PR TITLE
@craigspaeth: "fetchUntilEndInParallel"

### PR DESCRIPTION
**Pull request by dzucconi**
cc @mzikherman

### CODE REVIEW

File: lib/all.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import {
+  assign,
+  times,
+  flatten,
+} from 'lodash';
+import total from './loaders/total';
+import gravity from './loaders/gravity';
+
+export const all = (path, options = {}) => {
```
Comment (craigspaeth) [inline lib/all.js]: w00t w00t!

File: lib/loaders/gravity.js [hunk 2]
```diff
@@ -18,4 +19,6 @@ load.with = (accessToken) => {
   };
 };
 
+load.all = all;
```
Comment (craigspaeth) [inline lib/loaders/gravity.js]: Nice!

**Comment by craigspaeth** [comment]
Very nice! Much cleaner when you don't have to work with all that Backbone cruft!

File: lib/apis/fetch.js [hunk 3]
```diff
@@ -1,17 +1,27 @@
...
+      if (!!err || response.statusCode !== 200) {
+        return reject(err || new Error(response.body));
+      }
 
       try {
         const shouldParse = typeof response.body === 'string';
         const parsed = shouldParse ? JSON.parse(response.body) : response.body;
-        resolve(parsed);
+
+        resolve({
```
Comment (craigspaeth) [inline lib/apis/fetch.js]: Whoops! My bad for missing this, but it appears other places.. (atleast [here](https://github.com/artsy/metaphysics/blob/master/schema/me/bidder_positions.js#L29)) needs to be updated to accommodate this API change.

Comment (dzucconi) [inline lib/apis/fetch.js]: It shouldn't— the loader is still returning just the body.

Comment (craigspaeth) [inline lib/apis/fetch.js]: Weird—I am noticing that in most cases but in that case it's { body, headers}...

image

Comment (dzucconi) [inline lib/apis/fetch.js]: Ah, right me doesn't use the loader. I'll pr a fix in a bit

Comment (craigspaeth) [inline lib/apis/fetch.js]: Ah cool, thank you!!

**Pull request merged**